### ISSUE TITLE
Deprecate chef-rundeck

### ISSUE THREAD

**Issue by ev0ldave** [sous-chefs/rundeck#173]
@atheiman
Deprecate chef-rundeck as the project has gone stale and a better constructed project has evolved since then that is more flexible.
Possibly include support for better-chef-rundeck?  Would like to discuss moving this into it's own cookbook.

**Comment by atheiman** [comment]
i dont think that needs to be its own cookbook. at my office we run [better-chef-rundeck](https://github.com/atheiman/better-chef-rundeck) using [simple-passenger-cookbook](https://github.com/atheiman/simple-passenger-cookbook). all that is needed is to set a few attrs and `include_recipe 'simple_passenger::default'`

```ruby
# attributes.rb
default['passenger']['apps']['better-chef-rundeck'].tap do |bcr|
  bcr['git_repo'] = 'https://github.com/atheiman/better-chef-rundeck.git'
  bcr['git_revision'] = 'v0.5.0'
  bcr['passengerfile']['port'] = 8080
  bcr['passengerfile']['user'] = 'rundeck'
  bcr['passengerfile']['group'] = 'rundeck'
  bcr['ruby_bin'] = '/opt/chef/embedded/bin/ruby'
end

# recipe.rb
include_recipe 'simple_passenger'
```

**Comment by gsreynolds** [comment]
(Related: #140, #102)

**Comment by gsreynolds** [comment]
Quick brain dump/Note to self & others, for Chef 13 and for compatibility with how `chef-rundeck` translated chef node data to rundeck source config:

Until https://github.com/atheiman/better-chef-rundeck/pull/31 is merged, use ev0ldave branch chef13.
```ruby
# attributes.rb
default['passenger']['apps']['better-chef-rundeck'].tap do |bcr|
  bcr['git_repo'] = 'https://github.com/ev0ldave/better-chef-rundeck.git'
  bcr['git_revision'] = 'chef13'
  bcr['passengerfile']['port'] = 8080
  bcr['passengerfile']['user'] = 'rundeck'
  bcr['passengerfile']['group'] = 'rundeck'
  bcr['passengerfile']['envvars'] = { "BCR_CHEF_CONFIG": "/etc/chef/rundeck.rb" }
  bcr['ruby_bin'] = '/opt/chef/embedded/bin/ruby'
end
# recipe.rb
include_recipe 'simple_passenger'
```

By default `better-chef-rundeck` behaves differently to `chef-rundeck` in the attributes it returns, doesn't set hostname and nor a default username. Rundeck is unable to connect to remote nodes without these attributes.

`chef-rundeck` also translated the chef roles to rundeck tags, whereas `better-chef-rundeck` translates chef tags to rundeck tags for filtering and display in the UI (which is admittedly more correct but I'd argue less useful as at least from a personal point of view roles are more widely applied than tags in terms of grouping types of nodes).
```json
"project_settings": {
  "resources.source.1.config.url": "http://localhost:8080/*:*?environment=chef_environment&hostname=fqdn&ip=ipaddress&run_list=run_list&roles=roles&platform=platform&tags=roles&default_username=rundeck"
}
```
(for reference: https://github.com/atheiman/better-chef-rundeck/blob/83f7d9bfbaaaff9e310e21788b846c6c294f0e07/lib/better_chef_rundeck.rb)

**Comment by PowerSchill** [comment]
> Quick brain dump/Note to self & others from our wrapper cookbook, for Chef 13 and for compatibility with how `chef-rundeck` translated chef node data to rundeck source config:
> 
> Until [atheiman/better-chef-rundeck#31](https://github.com/atheiman/better-chef-rundeck/pull/31) is merged, use ev0ldave branch chef13.
> 
> ```ruby
> # attributes.rb
> default['passenger']['apps']['better-chef-rundeck'].tap do |bcr|
>   bcr['git_repo'] = 'https://github.com/ev0ldave/better-chef-rundeck.git'
>   bcr['git_revision'] = 'chef13'
>   bcr['passengerfile']['port'] = 8080
>   bcr['passengerfile']['user'] = 'rundeck'
>   bcr['passengerfile']['group'] = 'rundeck'
>   bcr['passengerfile']['envvars'] = { "BCR_CHEF_CONFIG": "/etc/chef/rundeck.rb" }
>   bcr['ruby_bin'] = '/opt/chef/embedded/bin/ruby'
> end
> # recipe.rb
> include_recipe 'simple_passenger'
> ```
> 
> By default `better-chef-rundeck` behaves differently to `chef-rundeck` in the attributes it returns, doesn't set hostname and nor a default username. Rundeck is unable to connect to remote nodes without these attributes.
> 
> `chef-rundeck` also translated the chef roles to rundeck tags for filtering and display in the UI, whereas `better-chef-rundeck` translates chef tags to rundeck tags (which is admittedly more correct but I'd argue less useful as at least from a personal point of view roles are more widely applied than tags in terms of grouping types of nodes).
> 
> ```json
> "project_settings": {
>   "resources.source.1.config.url": "http://localhost:8080/*:*?environment=chef_environment&hostname=fqdn&ip=ipaddress&run_list=run_list&roles=roles&platform=platform&tags=roles&default_username=rundeck"
> }
> ```
> 
> (for reference: https://github.com/atheiman/better-chef-rundeck/blob/83f7d9bfbaaaff9e310e21788b846c6c294f0e07/lib/better_chef_rundeck.rb)

Are you still using this solution? When using `simple_passenger` I encounter https://github.com/atheiman/simple-passenger-cookbook/issues/31

**Comment by damacus** [comment]
I think this seems like a sensible step and a pretty darn simple one. Once someone can work on the issue please go ahead!

**Comment by josephholsten** [comment]
FYI: @bscott is back at https://github.com/oswaldlabs/chef-rundeck/issues/71
